client/state/data-layer/wpcom/me/two-step/application-passwords/index.js

```diff
@@ -0,0 +1,55 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import deleteHandler from './delete';
+import newHandler from './new';
+import { APPLICATION_PASSWORDS_REQUEST } from 'state/action-types';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { mergeHandlers } from 'state/action-watchers/utils';
+import { receiveApplicationPasswords } from 'state/application-passwords/actions';
+
+/**
+ * Dispatches a request to fetch application passwords of the current user
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const requestApplicationPasswords = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: '/me/two-step/application-passwords',
+		},
+		action
+	);
+
+/**
+ * Dispatches a user application passwords receive action when the request succeeded.
+ *
+ * @param   {Object} action Redux action
+ * @param   {Object} data   Response from the endpoint
+ * @returns {Object} Dispatched user application passwords receive action
+ */
+export const handleRequestSuccess = ( action, { application_passwords: appPasswords } ) =>
+	receiveApplicationPasswords( appPasswords );
+
+const requestHandler = {
+	[ APPLICATION_PASSWORDS_REQUEST ]: [
+		dispatchRequestEx( {
```

Comment:
This, in particular, could benefit from a `fromApi` that validates the data at the edge as it enters the system.

Reply:
Good call! Introduced a `fromApi` simple validation in https://github.com/Automattic/wp-calypso/pull/22913/commits/6d163e990c750d4d8a578721213c5599c502b540